### ISSUE TITLE
pull using rebase always asks for branch

### ISSUE THREAD

**Issue by kev111** [akonwi/git-plus#436]
Any time I pull using rebase, it always ask for a branch with these options:

==Current==
origin/HEAD -> origin/master
origin/master

If I do it on the command line with git pull --rebase, it doesn't ask for the branch.  My remote seems to be set up properly.  Is there a way to fix this?

**Comment by alerque** [comment]
Same issue here. At the very least there should be a setting for whether to prompt for this or not. It's very difficult to train some of my non-coder writers and editors what to do when the workflow  (for Markdown prose publishing via git) includes prompts that they shouldn't have to do anything with.

**Comment by akonwi** [comment]
That feature allows pulling changes from other branches. I'm removing that `origin/HEAD` option because it's nearly equivalent to choosing `origin/master`. That way if you only have a single remote branch it will automatically pull from there. 

Otherwise selecting `==Current==` does a plain pull. That option was introduced so it wasn't necessary to think about where to pull from. #270

**Comment by alerque** [comment]
@akonwi Thanks for responding to this. And the thing you "fixed" is an improvement. The thing is it doesn't fix the problem for all use cases.

You're now calling it a feature, but in the use case for many of us out here this is a counter-productive feature. I'm sure there are people that need this, but the users I have setup with this package **need to not be prompted** for one more piece of information that they don't understand and can potentially break.

I might suggest a separate command for a pull with an option to rebase against a different branch. Or maybe a configuration option that says to always assume the current branch. There might be other solutions too.

Can this be re-opened pending a solution for this use case? Or should I open another issue?

**Comment by akonwi** [comment]
The next release will have an experimental toggle for this. Once we give it a go for about a week or so and there aren't any issues, I'll drop the experimental tag

**Comment by alerque** [comment]
Thanks for the effort on this @akonwi, this will make life easier for me, particularly is supporting users who can't actually get their heads around git. I'll be sure and check it out and report back if there are any issues with it in use.

**Comment by akonwi** [comment]
@alerque I'm assuming you haven't found any issues with the experimental feature. I've been using it as well but I would like validation from someone else before I make it a core feature

**Comment by alerque** [comment]
I have 3 (not git) people using it full time that make about 20 add/commit/pull/push executions a day and a half dozen others that use it sporadically. I haven't seen or heard of issues coming up for anyone that relate to this. Every one is appreciative of having one less step to remember, and the number of times I'm hearing "I thought I pushed that", "I don't see it yet", "Let me try again" has gone from about twice a day to zero.

**Comment by akonwi** [comment]
Awesome! I'll drop the experimental tag from that setting in the next release. Thanks!
